pydantic_ai_slim/pydantic_ai/agent/__init__.py

```
@@ -2579,6 +2604,86 @@ def _prepare_output_schema(self, output_type: OutputSpec[Any] | None) -> _output
+            )
+            tools_map = await toolset.get_tools(run_context)
+            tool_defs = [t.tool_def for t in tools_map.values()]
+
+            async def tool_runner(name: str, args: dict[str, Any], call_id: str) -> str:
+                tool = tools_map.get(name)
+                if tool is None:
+                    return f'Error: unknown tool {name!r}'
+                ctx = dataclasses.replace(run_context, tool_name=name, tool_call_id=call_id)
+                try:
+                    result = await toolset.call_tool(name, args, ctx, tool)
+                except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)